coderd/database/dump.sql

```diff
@@ -203,7 +203,8 @@ CREATE TABLE provisioner_daemons (
     created_at timestamp with time zone NOT NULL,
     updated_at timestamp with time zone,
     name character varying(64) NOT NULL,
-    provisioners provisioner_type[] NOT NULL
+    provisioners provisioner_type[] NOT NULL,
+    auth_token uuid
```

Comment:
What's the use-case for allowing `auth_token` to be NULL? Do we want to be able to revoke an auth token without deleting the provisioner daemon? Maybe we want to lookup what provisioner daemon had a specific auth token at some point? In that case it could make sense to have a `deleted bool` field instead of a nullable auth token.

Maybe auth tokens should be renewable too? That could also work as a delete/create using the same `name` (would also leave a "trace" about which auth token has produced what builds).

Another alternative would be to have `provisioner_daemon_auth_tokens` with fields like `daemon_id, token, granted, revoked`.

Just putting out some ideas, since I wasn't sure of the purpose of the nullability.

Reply:
Whoops, this could have used a better explanation. It doesn't have anything to do with changing or revoking auth tokens. 

With this change, we support both in-process and out-of-process provisioners, both of which are persisted in the database. An in-process provisioner has a NULL `auth_token` to represent the fact that external connections are not allowed to "become" that provisioner. That seemed cleaner than assigning a token that would never be used.